### PR TITLE
Add JS & CSS lint actions on PRs

### PR DESCRIPTION
**Pull request by RahiDroid**
## Description
Adds JS & CSS lint checks on PRs

## Fixes/Covers issue

Fixes #

### CODE REVIEW

File: .github/workflows/lint-js-css.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        run: npm ci
+
+      - name: JS Lint
+        run: npm run lint:js
+
+      - name: CSS Lint
+        run: npm run lint:css
+
+      - name: JS Lint
+        run: npm run lint:js
```
Comment (thelovekesh) [inline .github/workflows/lint-js-css.yml:39]:
```suggestion
```

File: .github/workflows/lint-js-css.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: Lint JS, CSS
+
+on:
+  pull_request:
+    paths:
```
Comment (thelovekesh) [inline .github/workflows/lint-js-css.yml:5]: Do we need to require paths explicitly? We already have specified paths in the `npm:lint*` commands.

**Review by thelovekesh — COMMENTED** [review]
I have a few more ideas in mind which can be implemented in this workflow:
- We can run linting for the package.json using `npm run lint:pkg-json`, so it's makes sense to add it as well.
- `actions/cache@v3` GHA can be used for node_modules cache.
- Add concurrency so that previous tests can be canceled if new changes pushed recently.
- We can specify when to run a workflow on PR i.e.
```yml
  pull_request:
    types:
      - synchronize
      - opened
      - ready_for_review
```
Should we run this workflow on Node 14 as well?

File: .github/workflows/lint-js-css.yml (hunk 1 above)
Comment (thelovekesh) [reply .github/workflows/lint-js-css.yml:39]: This is a duplicate entry.

**Comment by thelovekesh** [comment]
@RahiDroid Should we include JS tests in this workflow as well?

**Comment by RahiDroid** [comment]
> We can run linting for the package.json using npm run lint:pkg-json, so it's makes sense to add it as well.
Added!

> actions/cache@v3 GHA can be used for node_modules cache.
Actions documentation states that, "It is not recommended to cache node_modules, as it can break across Node versions and won’t work with npm ci".

Add concurrency so that previous tests can be canceled if new changes pushed recently.
> Added!

> We can specify when to run a workflow on PR 
Yup, adding `ready_for_review` makes sense, rest are default. Added!

> Should we run this workflow on Node 14 as well
I don't think it's necessary.

> @RahiDroid Should we include JS tests in this workflow as well?
Should be part of different workflow but will add in this PR

**Pull request merged**